### ISSUE TITLE
godot 2 -> 3 exporter change tilde characters (and probably other UTF-8 stuff)

### ISSUE THREAD

**Issue by Ranoller** [godotengine/godot#16850]
By petition in #16813

**Godot version:**

Godot 2.1 compiled at 18/02 and Godot 3 compiled in same day

**OS/device including version:**

Win 7 32 bit

**Issue description:**

I follow this steps:
1) I create new project in godot 2
2) I add in empty scene a Node2D and an AnimationPlayer
3) I click in "NewAnimation". New animation name in spanish is "Nueva Animación"
4) Save and use Godot 3 WIP exporter
5) Open project in Godot 3 and see that animation name is changed. Tilde is not respected.

![tildeexporterbug](https://user-images.githubusercontent.com/11712737/36416310-ff70dec4-1628-11e8-95ae-9b9b636f33f7.png)

**Minimal reproduction project:**

[TildeExporterProblem.zip](https://github.com/godotengine/godot/files/1739520/TildeExporterProblem.zip)

**Comment by Zireael07** [comment]
Probably related is #16838

**Comment by eon-s** [comment]
But the mentioned issue is about directories(where encoding may be different), this affects the exporter only.

tscn and tres files seem to be correctly saved on 2 and 3 but the exporter corrupts the characters on the exporting process(maybe using different encoding than text resource files).

**Comment by Ranoller** [comment]
This file is not recoverable (there is no program that can handle such huge string)... but please considerate this a bug like important-thing because can "silently" ruin some levels to someone.

**Comment by akien-mga** [comment]
The problem seems to be that in 2.1, the animation is referenced as:
```
"anims/Nueva animación" = SubResource( 1 )
```
while after conversion, it's:
```
anims/Nueva animación = SubResource( 1 )
```
I guess VariantParser doesn't handle it as a proper Unicode string without the quotes, and it gets corrupted each time you open the file in the editor and save it.

**Comment by akien-mga** [comment]
When saving an animation manually in Godot 3, the name is also properly escaped, so that's what needs to be fixed in the exporter:
```
"anims/Nueva animación 2" = SubResource( 1 )
anims/NuevaanimaciÃ³n = SubResource( 2 )
```

**Comment by akien-mga** [comment]
The bug also affects 3.0 itself, since it doesn't quote properties names without spaces, even if they contain non-ascii characters. So we can get e.g. this result:
```
"anims/Nueva animación 2" = SubResource( 1 )   // added manually in 3.0
anims/NuevaanimaciÃ^cÂ^cÃ^bÂ^cÃ^cÂ^bÃ^bÂ³n = SubResource( 2 )   // comes from 2 to 3 converter
anims/Testanim = SubResource( 3 )   // added manually in 3.0
anims/TestÃ³anim = SubResource( 4 )   // added manually in 3.0 as Testóanim
```

**Comment by akien-mga** [comment]
That same issue happens in 2.1 too if you create an animation without spaces but with non-ascii chars btw, which confirms that it's a manifestation of #6888.

**Comment by akien-mga** [comment]
Fixed by #17177.
